src/main/java/com/icthh/xm/uaa/security/UaaAuthenticationProvider.java

```diff
@@ -59,9 +71,25 @@ private AuthenticationProvider getProvider(Authentication authentication) {
     public Authentication authenticate(Authentication authentication) throws AuthenticationException {
         Authentication result = getProvider(authentication).authenticate(authentication);
         log.info("authenticated: {}, role: {}, {}",result.isAuthenticated(), result.getAuthorities(), result.getPrincipal());
+        checkPasswordExpiration(result);
+
         return result;
     }
 
+    private void checkPasswordExpiration(Authentication authentication) {
+        DomainUserDetails domainUserDetails = (DomainUserDetails) authentication.getPrincipal();
+        User user = userService.getUser(domainUserDetails.getUserKey());
+        LocalDate updatePasswordDate = user.getUpdatePasswordDate().atZone(UTC).toLocalDate();
+        Integer expirationPeriod = tenantPropertiesService.getTenantProps().getPasswordExpirationPeriod();
+        LocalDate currentDate = LocalDate.now();
+        log.info("check password expiration, passwordUpdateDate: {}, currentDate: {}, expirationPeriod: {}",
+            updatePasswordDate, currentDate, expirationPeriod);
+        long period =  DAYS.between(updatePasswordDate, currentDate);
+        if (period > expirationPeriod && expirationPeriod > 0) {
```

Comment:
Suggest to separate IF to check `expirationPeriod > 0` and perform all calculations and repository calls only if need:
```
private void checkPasswordExpiration(Authentication authentication) {
        Integer expirationPeriod = tenantPropertiesService.getTenantProps().getPasswordExpirationPeriod();
        if(expirationPeriod > 0) {
                User user = userService.getUser(domainUserDetails.getUserKey());
                  ....
                 if (period > expirationPeriod){
                            throw new CredentialsExpiredException("Password expiration period is over, please change password");
                 }
        }   

}
```